### PR TITLE
kubeadm:Clean up no longer used k8sVersion param

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
 cleanup for kubeadm.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:
/cc @luxas

**Release note**:

```release-note
NONE
```
~~`kubeadm join` now requires the `--discovery-token-ca-cert-hash` argument to be set, or the `--discovery-token-unsafe-skip-ca-verification` flag to be set for opting out of the CA pinning feature.~~

### CODE REVIEW

File: cmd/kubeadm/app/cmd/phases/bootstraptoken.go
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/bootstraptoken.go]: you can now remove this function completely as it's not needed anymore

File: cmd/kubeadm/app/cmd/phases/bootstraptoken.go
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/bootstraptoken.go]: these two lines can now be removed

File: cmd/kubeadm/app/cmd/init.go
Comment (luxas) [inline cmd/kubeadm/app/cmd/init.go]: you can now remove this call completely

File: cmd/kubeadm/app/apis/kubeadm/validation/validation.go
Comment (luxas) [inline cmd/kubeadm/app/apis/kubeadm/validation/validation.go]: the logic seems to change here? why add `len(cfg.DiscoveryToken) != 0` here?
cc @mattmoyer

File: cmd/kubeadm/app/apis/kubeadm/validation/validation.go
Comment (luxas) [inline cmd/kubeadm/app/apis/kubeadm/validation/validation.go]: "discovery-token-unsafe-skip-ca-verification"

**Review by luxas — COMMENTED** [review]
Some cleanup comments

File: cmd/kubeadm/app/apis/kubeadm/validation/validation.go
Comment (xiangpengzhao) [reply cmd/kubeadm/app/apis/kubeadm/validation/validation.go]: IIUC, only when the conditions has `len(cfg.DiscoveryToken) != 0` means "using token-based discovery" as is mentioned in the error message.

File: cmd/kubeadm/app/apis/kubeadm/validation/validation.go
Comment (xiangpengzhao) [reply cmd/kubeadm/app/apis/kubeadm/validation/validation.go]: Thanks @luxas! Will address this and other comments tomorrow :)

Comment (xiangpengzhao) [reply cmd/kubeadm/app/apis/kubeadm/validation/validation.go]: @luxas other comments addressed. But I'm a bit confusing here now. What were you expecting here to be? `allErrs = append(allErrs, field.Invalid(fldPath, "discovery-token-unsafe-skip-ca-verification",` or something else?

**Review by luxas — APPROVED** [review]
/lgtm
/approve

**Comment by luxas** [comment]
/hold

We need to update the e2e tests first: https://github.com/kubernetes/kubernetes-anywhere/blob/master/phase1/gce/configure-vm-kubeadm.sh#L106 in order to get a green run of the kubeadm e2e CI presubmit job that now is failing: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/54982/pull-kubernetes-e2e-kubeadm-gce/9008/

**Comment by xiangpengzhao** [comment]
@luxas PR https://github.com/kubernetes/kubernetes-anywhere/pull/483 sent. PTAL. Thanks!

**Comment by mattmoyer** [comment]
This overlaps with https://github.com/kubernetes/kubernetes/pull/55468 (pending approval).

**Comment by xiangpengzhao** [comment]
@mattmoyer thanks for pointing it out! I will rebase after #55468 merged. But I guess #55468 also needs to wait for https://github.com/kubernetes/kubernetes-anywhere/pull/483 being merged?

cc @luxas

**Review by luxas — APPROVED** [review]
/lgtm

**Pull request merged**
